### PR TITLE
GEOMESA-2178 Document inputs and outputs for GeoMesa WPS processes

### PR DESCRIPTION
**Pull request by rgrisot**
*Add parameters'name and description for all Geomesa processes
*Add example WPS request for HashProcess, KNNProcess, Point2PointProcess, ProximitySearchProcess, RouteSearchProcess, SamplingProcess, TrackLabelProcess and TubeSelectProcess
*Add response examples when it seemed relevant (when some extra fields are added to the response compared to a simple Query)

Signed-off-by: Rémi GRISOT <remi.grisot@atos.net>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
thanks for doing this! quick request before I review - could you make the example json output files much smaller (maybe ~100 lines each)? Git doesn't do especially well with large files like that.
Thanks,

File: docs/user/process.rst [hunk 1]
```diff
@@ -179,6 +496,69 @@ to collect. The available stat function are listed below:
...
+maxTime             Time as seconds for nofill & line gapfill methods.
+bufferSize          Buffer size in meters to use instead of maxSpeed/maxTime calculation.
+maxBins             Number of bins to use for breaking up query into individual queries.
+gapFill             Method of filling gap (nofill, line).
+=================   ===========
+
+.. _tubeSelectExampleXML:
+
+TubeSelect example (XML)
+""""""""""""""""""""""
```
Comment (elahrvivaz) [inline docs/user/process.rst]: title underline doesn't match title length

File: docs/user/process.rst [hunk 2]
```diff
@@ -179,6 +496,69 @@ to collect. The available stat function are listed below:
...
+==========  ===========
+data        Input features.
+track       Track attribute to use for grouping features.
+dtg         Date attribute to use for ordering tracks.
+==========  ===========
+
+.. _trackLabelExampleXML:
+
+TrackLabel example (XML)
+""""""""""""""""""""""
```
Comment (elahrvivaz) [inline docs/user/process.rst]: title underline doesn't match title length

File: docs/user/process.rst [hunk 3]
```diff
@@ -79,6 +117,285 @@ outputWidth   Width of the output raster in pixels.
 outputHeight  Height of the output raster in pixels.
 ============  ===========
 
+.. _date_offset_process:
+
+DateOffsetProcess
+^^^^^^^^^^^^^^
```
Comment (elahrvivaz) [inline docs/user/process.rst]: title underline doesn't match title length

File: docs/user/process.rst [hunk 4]
```diff
@@ -79,6 +117,285 @@ outputWidth   Width of the output raster in pixels.
...
+Hash example (XML)
+"""""""""""""""""""
+
+:download:`HashAttributeProcess_wps.xml </user/_static/process/HashProcess_wps.xml>` is a geoserver WPS call to the GeoMesa HashAttributeProcess. It can be run with the following curl call:
+
+.. code-block:: bash
+
+    curl -v -u admin:geoserver -H "Content-Type: text/xml" -d@HashAttributeProcess_wps.xml localhost:8080/geoserver/wps
+
+.. _queryExampleResults:
```
Comment (elahrvivaz) [inline docs/user/process.rst]: this is a duplicate tag - you might not need an anchor here if you're not referencing it anywhere

**Comment by rgrisot** [comment]
Thank you for the review: I reduced length of JSON files and fixed formatting issues in the process.rst file.
Sorry for the delay (I'm apprentice and I was not at my work until now) and the double commit: I saw your message about the formatting issues only after I pushed the reduced JSON examples. So now, pull request should have been updated.

File: docs/user/_static/process/HashProcess_results.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
+{
+      {
```
Comment (elahrvivaz) [inline docs/user/_static/process/HashProcess_results.json]: Looks like you trimmed off the opening line, so it's invalid json now. Could you add back in:
```
    "features" : [
```

File: docs/user/_static/process/Point2PointProcess_results.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
+{
+      {
```
Comment (elahrvivaz) [inline docs/user/_static/process/Point2PointProcess_results.json]: same missing line `"features" : [`

**Comment by elahrvivaz** [comment]
multiple commits are fine BTW, we always squash down to one commit during the merge.

**Comment by elahrvivaz** [comment]
Actually, I'll just go ahead and fix the json during the merge.

**Comment by elahrvivaz** [comment]
merged as 5d52ac2ae0268fa7039790e923baf0baf8dfb6a1

**Comment by elahrvivaz** [comment]
@rgrisot thanks again!

**Pull request closed without merging**